### ISSUE TITLE
Bootstrap 3.3.67 Collipsable panels conflict

### ISSUE THREAD

**Issue by Morzilla** [conversejs/converse.js#1440]
Hello everybody!
I'm trying to use ConverseJS overlayed chat into my website, but I have a problem with the Bootstrap Collipsable panels (I use the latest 3.3.7 bootstrap version): when I try to open or collpase a panel, something (often) it's reopening immediately without any animations, or some mess happens, like the text in the second panel that appears in overlay over the panel itself that is closed. If i comment out the ConverseJS javascript everything works fine...as soon as I re-insert that script the problem appears again. 

I used this example code for the collipsable panels:

`<div id="faq1" class="panel-group">
                                        <div class="panel panel-default">
                                            <div class="panel-heading">
                                                <h4 class="panel-title"><i class="fa fa-angle-right"></i> <a class="accordion-toggle" data-toggle="collapse" data-parent="#faq1" href="#faq1_q1">question 1?</a></h4>
                                            </div>
                                            <div id="faq1_q1" class="panel-collapse collapse in">
                                                <div class="panel-body">
                                                    <p>Lorem ipsum dolor sit amet.</p>
                                                </div>
                                            </div>
                                        </div>
                                        <div class="panel panel-default">
                                            <div class="panel-heading">
                                                <h4 class="panel-title"><i class="fa fa-angle-right"></i> <a class="accordion-toggle" data-toggle="collapse" data-parent="#faq1" href="#faq1_q2">question 2?</a></h4>
                                            </div>
                                            <div id="faq1_q2" class="panel-collapse collapse">
                                                <div class="panel-body">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas ultrices, justo vel imperdiet gravida, urna.</div>
                                            </div>
                                        </div>
                                        <div class="panel panel-default">
                                            <div class="panel-heading">
                                                <h4 class="panel-title"><i class="fa fa-angle-right"></i> <a class="accordion-toggle" data-toggle="collapse" data-parent="#faq1" href="#faq1_q3">Question 3?</a></h4>
                                            </div>
                                            <div id="faq1_q3" class="panel-collapse collapse">
                                                <div class="panel-body">Lorem ipsum dolor </div>
                                            </div>
                                        </div>
                                    </div>`

Do you have any hints? Thanks!

**Comment by jcbrand** [comment]
Are you sure it's the converse JavaScript and not the converse CSS?

Converse.js uses bootstrap 4 and uses [bootstrap.native](https://github.com/thednp/bootstrap.native/) for JS support for bootstrap CSS. However, in Boostrap 4 panels have been replaced with cards and I don't see anything in bootstrap.native that refers to "panels".

**Comment by Morzilla** [comment]
> Are you sure it's the converse JavaScript and not the converse CSS?
> 
> Converse.js uses bootstrap 4 and uses [bootstrap.native](https://github.com/thednp/bootstrap.native/) for JS support for bootstrap CSS. However, in Boostrap 4 panels have been replaced with cards and I don't see anything in bootstrap.native that refers to "panels".

Thanks for the reply! Yep, I'm sure, because if I comment out the CSS nothing changes (the problem is still present), but if I comment out the JS the problem is solved.

I use this version of converse.JS:

<script src="https://cdn.conversejs.org/4.1.0/dist/converse.min.js" charset="utf-8"></script>
 <link rel="stylesheet" type="text/css" media="screen" href="https://cdn.conversejs.org/4.1.0/css/converse.min.css" />

**Comment by jcbrand** [comment]
The core converse.js code won't be causing this problem because all the click events are scoped to be inside the `#conversejs` DOM element.

However, perhaps bootstrap.native has registered a handler that interferes.

Ideally we should find a way to scope bootstrap.native's code to only be active inside the #conversejs element as well. That might mean that we have to fork the code, but perhaps it can be done in a way that they'll merge our change back into the main branch.

**Comment by Echolon** [comment]
Is this still a valid issue?

**Comment by jcbrand** [comment]
Yes, this will still be a valid issue until we get rid of Bootstrap JavaScript for Converse or until we find a way to isolate the Converse DOM from any other Bootstrap code running in the page.

For the 2nd option, we can add a setting to enable the shadow DOM for the `converse-root` element. I think that could solve this issue.
